Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/DimensionSpacePoints.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of the Neos.ContentGraph.DoctrineDbalAdapter package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\ContentGraph\DoctrineDbalAdapter\Domain\Repository;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception\UniqueConstraintViolationException;
+use Neos\ContentRepository\Core\DimensionSpace\AbstractDimensionSpacePoint;
+use Neos\ContentRepository\Core\DimensionSpace\DimensionSpacePoint;
+use Neos\ContentRepository\Core\DimensionSpace\OriginDimensionSpacePoint;
+
+/**
+ * A place to cache dimensionspacepoint data, it's deterministic so there should never be an issue with having this in memory.
+ *
+ * @internal you should never need this in userland code
+ */
+final class DimensionSpacePoints
+{
+    /**
+     * @var string[]
+     */
+    private array $dimensionspacePoints = [];
+
+    public function __construct(
+        private readonly Connection $databaseConnection,
+        private readonly string $tableNamePrefix
+    ) {
+    }
+
+    public function insertDimensionSpacePoint(AbstractDimensionSpacePoint $dimensionSpacePoint): void
+    {
```

Comment:
IMO it would make sense to add the value to the runtime cache here, to prevent extra db lookups